test/unit/application_keys_test.rb

```diff
@@ -159,11 +158,10 @@ def test_format
   def test_regenerate_key
     @application_keys.add(key = 'app-key')
     @application.reload
+    Timecop.travel(2.hours.from_now)
     updated_at = @application.updated_at
     @application_keys.regenerate(key)
-    # @application.reload
     assert_not_equal updated_at, @application.updated_at
-
   end
```

Comment:
I propose something like this:

```ruby
def test_regenerate_key
  @application_keys.add(key = 'app-key')
  @application.reload
  updated_at = @application.updated_at
  Timecop.travel(2.hours.from_now) do
    @application_keys.regenerate(key)
  end
  assert_in_delta (updated_at + 2.hours),  @application.updated_at, 1.second
end
```

For the following reasons:
1. Using Timecop without a block and without a return in the teardown scares me because then if there is an error somewhere after the timecop, it might stay like that for the rest of the tests.
2. Using assert_not_equal or expects(:foo).never is ok when we do not really have a choice, but we can assert the exact value that we expect, it is better (less error-prone). Because otherwise it could pass the assert_not_equal for the wrong reasons (like 1 second of different when running the code)
3. Since we are gonna assert_equal, let's keep in mind the possible 1 second of margin of running the code from saving the previous updated_at to making the new update

Reply:
Agree! Though I barely touched this test, only make time travels.

About the teardown stuff 

https://github.com/3scale/porta/blob/e2dbb0cf87ed7a340261f681b9444254dd1a199b/test/test_helper.rb#L58-L61